test/gtest/uct/test_flush.cc

```diff
@@ -538,3 +539,265 @@ UCS_TEST_SKIP_COND_P(uct_flush_test, am_pending_flush_nb,
 }
 
 UCT_INSTANTIATE_TEST_CASE(uct_flush_test)
+
+class uct_cancel_test : public uct_test {
+    class peer {
+        uct_cancel_test &test;
+
+    public:
+
+        peer *m_peer;
+        entity *m_e;
+        mapped_buffer *m_buf1k;
+        mapped_buffer *m_buf8k;
+        mapped_buffer *m_buf32;
+
+        peer(uct_cancel_test *test, peer *peer) :
+            test(*test), m_peer(peer),
+            m_e(NULL), m_buf1k(NULL), m_buf8k(NULL), m_buf32(NULL) {}
+
+        void init() {
+            m_e = test.uct_test::create_entity(0, error_handler_cb);
+            test.m_entities.push_back(m_e);
+            m_buf1k = new mapped_buffer(1 * 1024, 0, *m_e);
+            m_buf8k = new mapped_buffer(8 * 1024, 0, *m_e);
+            m_buf32 = new mapped_buffer(32, 0, *m_e);
+            uct_iface_set_am_handler(m_e->iface(), 0, am_cb, &test,
+                                     UCT_CB_FLAG_ASYNC);
+        }
+
+        void cleanup() {
+            if (m_buf1k) {
+                delete m_buf1k;
+                delete m_buf8k;
+                delete m_buf32;
+                m_buf1k = NULL;
+            }
+        }
+
+        ~peer() {
+            cleanup();
```

Comment:
why needed ? 
if really needed use ucs::auto_ptr

Reply:
need to call it before uct_test::cleanup() in regular flow and during destruction in skip flow